plugins/woocommerce/includes/shipping/flat-rate/class-wc-shipping-flat-rate.php

```diff
@@ -281,7 +281,7 @@ public function find_shipping_classes( $package ) {
 	public function sanitize_cost( $value ) {
 		$value = is_null( $value ) ? '' : $value;
 		$value = wp_kses_post( trim( wp_unslash( $value ) ) );
-		$value = str_replace( array( get_woocommerce_currency_symbol(), html_entity_decode( get_woocommerce_currency_symbol() ), wc_get_price_thousand_separator() ), '', $value );
+		$value = str_replace( array( get_woocommerce_currency_symbol(), html_entity_decode( get_woocommerce_currency_symbol() ) ), '', $value );
```

Comment:
This was interfering with saving an "invalid" value, in essence trying to outsmart the user:

1. Set thousands separator to `.`
2. Set decimal separator to `,`
3. Attempt to save `7.5`
4. Now it should still save `7.5` instead of `75`